src/lib.rs

```diff
@@ -212,6 +212,62 @@ impl<I, P> FixedPoint<I, P> {
     }
 }
 
+/// Avoids the use of [`std::io::Cursor`] in `no_std` context
+struct Cursor<T> {
```

Comment:
`lib.rs` is big enough, I suggest moving this struct and new functions below to `no_std.rs` or `std_shim.rs`